include/osquery/core.h

```diff
@@ -92,10 +92,47 @@ osquery::QueryData query(const std::string& q, int& error_return, sqlite3* db);
  * virtual tables attached. This method is the main abstraction for creating
  * SQLite3 databases within osquery.
  *
+ * Note: osquery::initOsquery must be called before calling createDB in order
+ * for virtual tables to be registered.
+ *
  * @return a SQLite3 database with all virtual tables attached
  */
 sqlite3* createDB();
 
+
+/**
+ * @brief Analyze a query, returning information about the result columns
+ *
+ * This function asks SQLite to determine what the names and types are of the
+ * result columns of the provided query. Only table columns (not expressions or
+ * subqueries) can have their types determined. Types that are not determined
+ * are indicated with the string "UNKNOWN".
+ *
+ * @param q the query to analyze
+ * @param error_return an int indicating success or failure of analysis
+ *
+ * @return a vector containing pairs of <name, type> for each result column
+ */
+std::vector<std::pair<std::string, std::string> >
+  getQueryColumns(const std::string& q, int& error_return);
```

Comment:
What do you think about returning a `osquery::Status` and filling in the vector though a parameter output?

There's https://github.com/facebook/osquery/blob/master/include/osquery/tables.h#L70:
```cpp
typedef const std::vector<std::pair<std::string, std::string> > TableColumns;
```